javaagent/Dockerfile

```
@@ -2,5 +2,6 @@ FROM eclipse-temurin:11-jre
 
 ADD build/libs/app.jar /app.jar
 ADD build/agent/opentelemetry-javaagent.jar /opentelemetry-javaagent.jar
+ADD build/agent/opentelemetry-javaagent-extension.jar /opentelemetry-javaagent-extension.jar
 
-ENTRYPOINT java -jar -javaagent:/opentelemetry-javaagent.jar /app.jar
+ENTRYPOINT java -jar -javaagent:/opentelemetry-javaagent.jar -Dotel.javaagent.extensions=/opentelemetry-javaagent-extension.jar /app.jar
```

Loading the rule based sampler artifact as an agent extension, so it can be referenced in the declarative config file.

---

when i was running this locally i had issues with the ordering of -jar -javaagent
```suggestion
ENTRYPOINT java -javaagent:/opentelemetry-javaagent.jar -jar -Dotel.javaagent.extensions=/opentelemetry-javaagent-extension.jar /app.jar
```

---

Weird.. I wonder why I didn't.. I moved `-jar` all the way to the end, right before `/app.jar`.